nixos/modules/services/networking/dnscrypt-proxy.nix

```diff
@@ -277,7 +277,8 @@ in
         ExecStart = "${dnscrypt-proxy}/bin/dnscrypt-proxy ${toString daemonArgs}";
         Restart = "always";
         User = "dnscrypt-proxy";
-
+        CapabilityBoundingSet = "CAP_NET_BIND_SERVICE";
```

Comment:
But AmbientCapabilities *grants* the capability to an unprivileged process; with this patch, the service runs with more caps than before.

For this to be worth it we'd have to expect that systemd would somehow decide to run the daemon as root and still respect the caps. That sounds like a serious bug in systemd to me and I think it's inappropriate to attempt to account for it. I mean, if we start trying to anticipate odd bugs like that, we're entering a fairly deep rabbit hole ...